### PR TITLE
feat(module:water-mark): nzWaterMark is a block element && standalone

### PR DESCRIPTION
**Pull request by OriginRing**
fix: #8187

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: components/water-mark/water-mark.component.ts [hunk 1]
```diff
@@ -120,7 +122,8 @@ export class NzWaterMarkComponent implements AfterViewInit, OnInit, OnChanges, O
       width: '100%',
       height: '100%',
       pointerEvents: 'none',
-      backgroundRepeat: 'repeat'
+      backgroundRepeat: 'repeat',
+      visibility: 'visible !important'
```
Comment (HyperLife1119) [inline components/water-mark/water-mark.component.ts:126]: 我们可以避免使用 `!important` 吗？

File: components/water-mark/water-mark.component.ts [hunk 2]
```diff
@@ -30,6 +30,8 @@ const FontGap = 3;
 
 @Component({
   selector: 'nz-water-mark',
+  standalone: true,
+  imports: [CommonModule],
```
Comment (HyperLife1119) [inline components/water-mark/water-mark.component.ts:34]: 看起来我们并没有使用到 CommonModule，我们可以删除它 :)

Comment (OriginRing) [reply components/water-mark/water-mark.component.ts:34]: 是的，我处理一下

**Review by HyperLife1119 — APPROVED** [review]
LGTM. WDYT? @Laffery

**Pull request merged**